File: 정수빈/Programmers/Level2/숫자의표현.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12924

def solution(n):
    answer = 0
    for i in range(1, 1+n):
        for j in range(i, 1+n):
            s = sum(range(i, j+1))
            if s > n:
                break
            elif s == n:
                answer +=1            
    return answer

File: 장세훈/Programmers/Level2/숫자의표현.java
class 숫자의표현 {

    /*
        https://school.programmers.co.kr/learn/courses/30/lessons/12924
     */

    public int solution(int n) {
        int answer = 0;
        int[] arr = new int[n + 1];

        for (int i = 0; i < arr.length; i++) {
            arr[i] = i;
        }

        int l = 1;
        int r = 1;
        int sum = 1;

        while (l <= r && l <= n && r <= n) {
            if (sum >= n) {
                if (sum == n) answer++;
                sum -= arr[l++];
            } else {
                sum += arr[++r];
            }
        }

        return answer;
    }
}